### PR TITLE
Redesign Filing Landing Page + Keycloak upgrade to version 25

### LINKED ISSUE #2162: Redesign Filing Landing Page
To Do:
- [x] Add Keycloak Login EOL Notice
- [x] Add relevant filing info to page:
- -  [x] Link to Online FIG 
- -  [x] Filing Deadlines

hmda-filing-before

### PR DESCRIPTION
**Pull request by Michaeldremy**
Closes #2162 & #2319

### Pull request includes:

- Redesigned Filing Landing Page
- `keycloaks-js` was upgraded/migrated from version 11 to version 25
- Adjusted max-width to align with rest of the website in the Filing platform
- Updated Cypress Tests
   - Updated Filing tests to utilize the year selector and filing period selectors instead of relying on the URL
   - HMDA Help test that deleted and created an institution has been disabled
- HMDA Help `Quarterly Filer` select has been disabled and is no longer able to be interacted with 

### Before Redesign

image

### After Redesign

image

### CODE REVIEW

File: src/common/constants/dev-config.json [hunk 1]
```diff
@@ -44,10 +43,16 @@
...
+      "Q1": "04/01/2026 - 05/30/2026 - 06/30/2026",
+      "Q2": "07/01/2026 - 08/29/2026 - 09/30/2026",
+      "Q3": "10/01/2026 - 11/30/2026 - 12/31/2026",
+      "annual": "01/01/2027 - 03/01/2027 - 12/31/2029"
+    },
     "2025": {
       "Q1": "04/01/2025 - 05/30/2025 - 06/30/2025",
       "Q2": "07/01/2025 - 08/29/2025 - 09/30/2025",
-      "Q3": "10/01/2025 - 11/29/2025 - 12/31/2025",
+      "Q3": "10/01/2025 - 12/01/2025 - 12/31/2025",
```
Comment (Michaeldremy) [inline src/common/constants/dev-config.json:55]: If there is any questions reference ticket #2301 to see why the date is being changed.

File: src/filing/institutions/InstitutionPeriodSelector.jsx [hunk 2]
```diff
@@ -31,6 +31,8 @@ const InstitutionPeriodSelector = ({
           dispatch(updateFilingPeriod(opt.value))
           history.replace(pathname.replace(filingPeriod, opt.value))
         }}
+        className='filing-year-selector'
+        classNamePrefix='filing-year'
```
Comment (Michaeldremy) [inline src/filing/institutions/InstitutionPeriodSelector.jsx:35]: Used to properly select the dropdown for Cypress testing.

File: src/filing/institutions/InstitutionPeriodSelector.jsx [hunk 3]
```diff
@@ -44,6 +46,7 @@ const InstitutionPeriodSelector = ({
             history.replace(pathname.replace(filingPeriod, period))
           }}
           isDisabled={quarterOpts.length < 2}
+          classNamePrefix='annual-or-quarter'
```
Comment (Michaeldremy) [inline src/filing/institutions/InstitutionPeriodSelector.jsx:49]: Used to properly select the dropdown for Cypress testing.

**Review by ojbravo — APPROVED** [review]
Amazing work!!!!

**Pull request merged**